src/console/text_printer.rs

```diff
@@ -0,0 +1,97 @@
+use crate::console::Printer;
+use std::fmt;
+use std::io::Write;
+
+pub struct Group {
+    collapsed: bool,
+}
+
+pub(crate) struct TextPrinter<W: Write> {
+    writer: W,
+    groups: Vec<Group>,
+}
+
+impl<W: Write> TextPrinter<W> {
+    pub fn new(writer: W) -> TextPrinter<W> {
+        TextPrinter {
+            writer,
+            groups: vec![],
+        }
+    }
+}
+
+impl<W: Write> Printer for TextPrinter<W> {
+    fn print(&mut self, log_level: &str, args: &[&dyn fmt::Display], _options: &[&str]) {
```

Comment:
```suggestion
    fn print(&mut self, log_level: LogLevel, args: &[&dyn fmt::Display], _options: &[&str]) {
```

Here `LogLevel` is the new enum mentioned above.